packages/11ty/_plugins/figures/index.js

```diff
@@ -0,0 +1,17 @@
+const Figure = require('./figure')
+const iiifConfig = require('./config')
+
+module.exports = function (eleventyConfig, options = {}) {
+  eleventyConfig.addGlobalData('iiifConfig', iiifConfig(eleventyConfig))
+
+  eleventyConfig.on('eleventy.before', async () => {
+    let { figure_list: figureList } = eleventyConfig.globalData.figures
+    figureList = await Promise.all(
+      figureList.map((data) => {
+        const figure = new Figure(eleventyConfig, data)
+        return figure.create()
+      })
+    )
+    Object.assign(eleventyConfig.globalData.figures.figure_list, figureList)
+  })
```

Comment:
Is renaming `figures_list` as `figures` confusing here? It would be nice to be able to eventually drop the additional `*_list` keys from the yaml arrays.
```suggestion
  eleventyConfig.on('eleventy.before', async () => {
    let { figure_list: figures } = eleventyConfig.globalData.figures
    figures = await Promise.all(
      figures((data) => {
        const figure = new Figure(eleventyConfig, data)
        return figure.create()
      })
    )
    Object.assign(eleventyConfig.globalData.figures.figure_list, figures)
  })
```